static/app/components/searchQueryBuilder/index.tsx

```diff
@@ -143,31 +147,32 @@ function SearchIndicator({
   );
 }
 
-const ActionButtons = forwardRef<HTMLDivElement, {trailingItems?: React.ReactNode}>(
-  ({trailingItems = null}, ref) => {
-    const {dispatch, handleSearch, disabled} = useSearchQueryBuilder();
-
-    if (disabled) {
-      return null;
-    }
+const ActionButtons = forwardRef<
+  HTMLDivElement,
+  {hideClearWhenEmpty?: boolean; trailingItems?: React.ReactNode}
+>(({trailingItems = null, hideClearWhenEmpty = false}, ref) => {
+  const {dispatch, handleSearch, disabled, query} = useSearchQueryBuilder();
 
-    return (
-      <ButtonsWrapper ref={ref}>
-        {trailingItems}
-        <ActionButton
-          aria-label={t('Clear search query')}
-          size="zero"
-          icon={<IconClose />}
-          borderless
-          onClick={() => {
-            dispatch({type: 'CLEAR'});
-            handleSearch('');
-          }}
-        />
-      </ButtonsWrapper>
-    );
+  if (disabled || (hideClearWhenEmpty && query === '')) {
```

Comment:
We still want to render `trailingItems` when the query is empty

Reply:
Sounds good! i'll remove the prop and still render trailing items 👍